### PR TITLE
Support for mabl test branch

### PR DESCRIPTION
**Pull request by bertold**
Added support for specifying the mabl test branch
Updated mockito dependency and marked it as a test-only dependency
Replaced a deprecated call to the serialization library

### CODE REVIEW

File: src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/config.jelly [hunk 1]
```diff
@@ -29,6 +29,9 @@
         </f:description>
       </j:if>
     </f:entry>
+    <f:entry title="${%Test branch}" field="mablBranch">
```
Comment (gone-skiing) [inline src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/config.jelly:32]: I wonder if this should be mabl branch. As opposed to source branch. We also branch flows and other things. I am not 100% sure what the best term here would be.

Comment (bertold) [reply src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/config.jelly:32]: I'll change this to mabl branch, thank you.

File: src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java [hunk 2]
```diff
@@ -71,7 +71,7 @@
         objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
         objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
         objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
-        objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
+        objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java:74]: Thanks for clearing the deprecation warning.

File: src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/help-mablBranch.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<div>
+    mabl will run tests using this branch name
```
Comment (twistedpair) [inline src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/help-mablBranch.html:2]: Perhaps we should clarify this is a `mabl` branch, not an SCM branch?

Comment (bertold) [reply src/main/resources/com/mabl/integration/jenkins/MablStepBuilder/help-mablBranch.html:2]: Absolutely, thank you.

**Pull request merged**